search.go

```diff
@@ -124,6 +125,10 @@ func (c *Client) Search(ctx context.Context, query string, t SearchType, opts ..
 	v.Set("q", query)
 	v.Set("type", t.encode())
 
+	if t == SearchTypeShow && v.Get("market") == "" {
+		return nil, fmt.Errorf("searching for shows requires market parameter")
```

Comment:
maybe worth creating a new error type if it doesn't already exist like 
```go
var ErrSearchMarketRequired = errors.New("spotify::search: searching for shows requires market parameter")
```

Reply:
Agree with this and will include it as part of PR.